$NetBSD: patch-ab,v 1.11 2016/02/04 11:50:24 leot Exp $

o Handle the various operating systems in the same way avoiding hardcoding
o Build and link with openjpeg-2.1

--- Makerules.orig	2015-11-10 16:19:51.000000000 +0000
+++ Makerules
@@ -29,46 +29,6 @@ ifeq "$(largefile)" "yes"
 CFLAGS += -DFZ_LARGEFILE
 endif
 
-# Windows (MINGW) build doesn't use system libraries.
-ifeq "$(OS)" "MINGW"
-
-HAVE_X11 ?= no
-HAVE_GLUT ?= yes
-
-SYS_GLUT_CFLAGS = -DFREEGLUT_STATIC -Ithirdparty/freeglut/include
-SYS_GLUT_LIBS = -Lthirdparty/freeglut/lib -lfreeglut_static -lopengl32 -lgdi32 -lwinmm
-
-# Mac OS X doesn't have pkg-config so we hard code paths.
-else ifeq "$(OS)" "MACOS"
-
-HAVE_X11 ?= yes
-HAVE_GLUT ?= yes
-
-SYS_OPENSSL_CFLAGS = -DHAVE_OPENSSL
-SYS_OPENSSL_LIBS = -lcrypto
-
-SYS_CURL_DEPS = -lpthread
-
-SYS_X11_CFLAGS = -I/usr/X11R6/include
-SYS_X11_LIBS = -L/usr/X11R6/lib -lX11 -lXext
-SYS_GLUT_CFLAGS = -Wno-deprecated-declarations
-SYS_GLUT_LIBS = -framework GLUT -framework OpenGL
-
-SYS_FREETYPE_CFLAGS = -I/usr/X11R6/include/freetype2
-SYS_FREETYPE_LIBS = -lfreetype
-SYS_OPENJPEG_LIBS = -lopenjpeg
-SYS_JBIG2DEC_LIBS = -ljbig2dec
-SYS_JPEG_LIBS = -ljpeg
-SYS_ZLIB_LIBS = -lz
-
-CC = xcrun cc
-AR = xcrun ar
-LD = xcrun ld
-RANLIB_CMD = xcrun ranlib $@
-
-# Linux uses pkg-config for system libraries.
-else ifeq "$(OS)" "Linux"
-
 HAVE_X11 ?= yes
 HAVE_GLUT ?= yes
 
@@ -87,8 +47,16 @@ SYS_CURL_DEPS = -lpthread -lrt
 SYS_X11_CFLAGS = $(shell pkg-config --cflags x11 xext)
 SYS_X11_LIBS = $(shell pkg-config --libs x11 xext)
 
-SYS_GLUT_CFLAGS =
-SYS_GLUT_LIBS = -lglut -lGL
+ifeq "$(shell pkg-config --exists glut && echo yes)" "yes"
+SYS_GLUT_CFLAGS = $(shell pkg-config --cflags glut)
+SYS_GLUT_LIBS = $(shell pkg-config --libs glut)
+endif
+
+ifeq "$(shell pkg-config --exists glfw3 && echo yes)" "yes"
+HAVE_GLFW = yes
+SYS_GLFW_CFLAGS = $(shell pkg-config --cflags glfw3)
+SYS_GLFW_LIBS = $(shell pkg-config --libs glfw3)
+endif
 
 SYS_FREETYPE_CFLAGS = $(shell pkg-config --cflags freetype2)
 SYS_FREETYPE_LIBS = $(shell pkg-config --libs freetype2)
@@ -98,8 +66,6 @@ SYS_JBIG2DEC_LIBS = -ljbig2dec
 SYS_JPEG_LIBS = -ljpeg
 SYS_ZLIB_LIBS = -lz
 
-endif
-
 # The following section is an example of how to simply do cross-compilation
 # using these Makefiles. It builds for a beagleboard running ARM linux,
 # compiling on windows with the CodeSourcery G++ compilers.
